### ISSUE TITLE
SSH to remote system - Not working

### ISSUE THREAD

**Issue by masongk** [microsoft/vscode-remote-release#5536]
Does this issue occur when all extensions are disabled?: No

- VS Code Version: 1.59.1
- OS Version: Windows 10
- Remote - SSH Editing Configuration Files - v0.65.7
- Remote - SSH - v0.65.7

Steps to Reproduce:

1. When trying to SSH into the remote target using the ssh, got the error 

> channel 3: open failed: unknown channel type: unknown channel type
channel 4: open failed: unknown channel type: unknown channel type
2. A remote system is not connected to internet. I am able to ssh from putty. Attaching the log for additional infomration

> alphauser@172.17.2.2's password: 
3dbc7a40ccca: running
Acquiring lock on /home/alphauser/.vscode-server/bin/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa/vscode-remote-lock.alphauser.3866c3553be8b268c8a7f8c0482c0c0177aa8bfa
Installing to /home/alphauser/.vscode-server/bin/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa...
3dbc7a40ccca%%1%%
Downloading with wget
Download complete
3dbc7a40ccca%%2%%
tar --version:
tar (GNU tar) 1.29
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

>Written by John Gilmore and Jay Fenlason.
Checking /home/alphauser/.vscode-server/.3866c3553be8b268c8a7f8c0482c0c0177aa8bfa.log and /home/alphauser/.vscode-server/.3866c3553be8b268c8a7f8c0482c0c0177aa8bfa.pid for a running server
Starting server with command... /home/alphauser/.vscode-server/bin/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa/server.sh --start-server --host=127.0.0.1 --enable-remote-auto-shutdown  --port=0  &> "/home/alphauser/.vscode-server/.3866c3553be8b268c8a7f8c0482c0c0177aa8bfa.log" < /dev/null
printenv:
    LC_ALL=en_US.UTF-8
    SSH_CONNECTION=10.44.108.56 36971 172.17.2.2 22
    LANG=en_US.UTF-8
    XDG_SESSION_ID=1306
    USER=alphauser
    PWD=/home/alphauser
    HOME=/home/alphauser
    SSH_CLIENT=10.44.108.56 36971 22
    https_proxy=http://127.0.0.1:3128/
    http_proxy=http://127.0.0.1:3128/
    MAIL=/var/mail/alphauser
    SHELL=/bin/bash
    socks_proxy=socks://127.0.0.1:3128/
    SHLVL=2
    VSCODE_AGENT_FOLDER=/home/alphauser/.vscode-server
    LOGNAME=alphauser
    DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1049/bus
    XDG_RUNTIME_DIR=/run/user/1049
    PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
    ftp_proxy=http://127.0.0.1:3128/
    _=/usr/bin/printenv
    OLDPWD=/home/alphauser/.vscode-server/bin/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa
rm: cannot remove '/home/alphauser/.vscode-server/.3866c3553be8b268c8a7f8c0482c0c0177aa8bfa.token': No such file or directory
Spawned remote server: 54465
Waiting for server log...
Waiting for server log...
 

> Reminder: You may only use this software with Visual Studio family products,
 as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)

>3dbc7a40ccca: start
sshAuthSock====
webUiAccessToken====
listeningOn==46641==
osReleaseId==ubuntu==
arch==x86_64==
tmpDir==/run/user/1049==
platform==linux==
unpackResult==success==
didLocalDownload==0==
downloadTime==12241==
installTime==936==
extInstallTime====
serverStartTime==86==
connectionToken==584c96db-2987-4c15-b9ae-1823aa85395e==
3dbc7a40ccca: end
channel 3: open failed: unknown channel type: unknown channel type
channel 4: open failed: unknown channel type: unknown channel type

**Comment by roblourens** [comment]
Please share the log from the Remote-SSH output channel

**Comment by masongk** [comment]
> [12:00:46.972] Log Level: 2
[12:00:46.977] remote-ssh@0.65.7
[12:00:46.977] win32 x64
[12:00:46.978] SSH Resolver called for "ssh-remote+172.17.2.2", attempt 1
[12:00:46.978] "remote.SSH.useLocalServer": false
[12:00:46.978] "remote.SSH.showLoginTerminal": true
[12:00:46.978] "remote.SSH.remotePlatform": {"172.17.2.2":"linux"}
[12:00:46.978] "remote.SSH.path": undefined
[12:00:46.978] "remote.SSH.configFile": undefined
[12:00:46.978] "remote.SSH.useFlock": true
[12:00:46.978] "remote.SSH.lockfilesInTmp": false
[12:00:46.979] "remote.SSH.localServerDownload": auto
[12:00:46.979] "remote.SSH.remoteServerListenOnSocket": false
[12:00:46.979] "remote.SSH.showLoginTerminal": true
[12:00:46.979] "remote.SSH.defaultExtensions": []
[12:00:46.979] "remote.SSH.loglevel": 2
[12:00:46.979] SSH Resolver called for host: 172.17.2.2
[12:00:46.979] Setting up SSH remote "172.17.2.2"
[12:00:47.007] Using commit id "3866c3553be8b268c8a7f8c0482c0c0177aa8bfa" and quality "stable" for server
[12:00:47.009] Install and start server if needed
[12:00:47.016] Checking ssh with "ssh -V"
[12:00:47.064] > OpenSSH_for_Windows_8.1p1, 
[12:00:47.064] > LibreSSL 3.0.2

>[12:00:47.068] Running script with connection command: ssh -T -D 56389 "172.17.2.2" bash
[12:00:47.070] Terminal shell path: C:\Windows\System32\cmd.exe
[12:00:47.350] > ]0;C:\Windows\System32\cmd.exe
[12:00:47.350] Got some output, clearing connection timeout
[12:00:47.368] > 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
>[12:00:49.785] > alphauser@172.17.2.2's password: 
[12:00:56.979] > 
[12:00:59.181] > bf86ebce4552: running
[12:00:59.574] > Acquiring lock on /home/alphauser/.vscode-server/bin/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa/vscode-remote-lock.alphauser.3866c3553be8b268c8a7f8c0482c0c0177aa8bfa
> Found existing installation at /home/alphauser/.vscode-server/bin/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa...
> Checking /home/alphauser/.vscode-server/.3866c3553be8b268c8a7f8c0482c0c0177aa8bfa.log and /home/alphauser/.vscode-server/.3866c3553be8b268c8a7f8c0482c0c0177aa8bfa.pid for a running server
> Looking for server with pid: 54465
> Found running server...
> 
> 
>  Reminder: You may only use this software with Visual Studio family products,
>  as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> 
>[12:00:59.593] > 
>12:00:59.757] >  
> Checking server status on port 46641 with wget
> bf86ebce4552: start
> SSH_AUTH_SOCK====
> DISPLAY====
> webUiAccessToken====
> listeningOn==46641==
> osReleaseId==ubuntu==
> arch==x86_64==
> tmpDir==/run/user/1049==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
>[12:00:59.762] > 
> extInstallTime====
> serverStartTime====
> connectionToken==111a11aa-1111-1a11-a1aa-1111aa11111a==
> bf86ebce4552: end
>[12:00:59.762] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
webUiAccessToken====
listeningOn==46641==
osReleaseId==ubuntu==
arch==x86_64==
tmpDir==/run/user/1049==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime====
connectionToken==111a11aa-1111-1a11-a1aa-1111aa11111a==

>[12:00:59.763] Remote server is listening on 46641
[12:00:59.763] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":46641},"osReleaseId":"ubuntu","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/run/user/1049","platform":"linux","connectionToken":"111a11aa-1111-1a11-a1aa-1111aa11111a"},"installUnpackCode":""}
[12:00:59.764] Starting forwarding server. localPort 56399 -> socksPort 56389 -> remotePort 46641
[12:00:59.765] Forwarding server listening on 56399
[12:00:59.765] Waiting for ssh tunnel to be ready
[12:00:59.766] Tunneled 46641 to local port 56399
[12:00:59.766] Resolved "ssh-remote+172.17.2.2" to "127.0.0.1:56399"
[12:00:59.767] [Forwarding server 56399] Got connection 0
[12:00:59.770] ------




>[12:00:59.776] [Forwarding server 56399] Got connection 1
[12:00:59.961] Failed to set up socket for dynamic port forward to remote port 46641: Socket closed. Is the remote port correct?
[12:00:59.967] > channel 3: open failed: unknown channel type: unknown channel type
[12:00:59.968] Failed to set up socket for dynamic port forward to remote port 46641: Socket closed. Is the remote port correct?
[12:00:59.975] > channel 4: open failed: unknown channel type: unknown channel type
[12:00:59.983] >

**Comment by roblourens** [comment]
Sorry, I have no idea where the `unknown channel type` message comes from. Basically the port forwarding is failing. This was reported once before and was not solved: https://github.com/microsoft/vscode-remote-release/issues/326

Maybe some corporate policy interfering with port forwarding?

**Comment by masongk** [comment]
Ok.. I have also searched in other forums. I followed 2 approaches which seems to have resolved the issue for other people.

I'm recording it here so it helps others.
- Restarting the remote server (resolved the issue for some people).
- Stopping all the ~~.vscode~~ process in remote server and deleting ~~/home/$USER/.vscode~~ folder resolved the issue for my colleague.

Unfortunately, it didn't work for me and restarting the remote server is not under my control. Corporate policy 😅.
